clients/go/test/api_locales_test.go

```diff
@@ -0,0 +1,157 @@
+
+		// require.Nil(t, err)
+		// require.NotNil(t, resp)
+		// assert.Equal(t, 200, httpRes.StatusCode)
+
+	})
+
+	t.Run("Test LocalesApiService LocaleCreate", func(t *testing.T) {
+
+		// t.Skip("skip test") // remove to run test
+
+		// var projectId string
+
+		// resp, httpRes, err := apiClient.LocalesApi.LocaleCreate(context.Background(), projectId).Execute()
+
+		// require.Nil(t, err)
+		// require.NotNil(t, resp)
+		// assert.Equal(t, 200, httpRes.StatusCode)
+
+	})
+
+	t.Run("Test LocalesApiService LocaleDelete", func(t *testing.T) {
+
+		// t.Skip("skip test") // remove to run test
+
+		// var projectId string
+		// var id string
+
+		// httpRes, err := apiClient.LocalesApi.LocaleDelete(context.Background(), projectId, id).Execute()
+
+		// require.Nil(t, err)
+		// assert.Equal(t, 200, httpRes.StatusCode)
+
+	})
+
+	t.Run("Test LocalesApiService LocaleDownload", func(t *testing.T) {
+		formatOptions := optional.NewInterface(map[string]interface{}{
+			"foo": "bar",
+			"baz": "bazz",
+		})
+		localeDownloadOpts := phrase.LocaleDownloadOpts{FormatOptions: formatOptions}
+
+		resp, httpRes, err := apiClient.LocalesApi.LocaleDownload(context.Background(), "project_id_example", "locale_id", &localeDownloadOpts)
+		requestUrl := httpRes.Request.URL
+		
+
+		require.Nil(t, err)
+		require.NotNil(t, resp)
+		assert.Equal(t, 200, httpRes.StatusCode)
+		assert.Equal(t, "format_options%5Bbaz%5D=bazz&format_options%5Bfoo%5D=bar", requestUrl.RawQuery)
+		assert.Equal(t, "/projects/project_id_example/locales/locale_id/download", requestUrl.Path)
+		assert.Equal(t, "GET", httpRes.Request.Method)
+	})
+
+	t.Run("Test LocalesApiService LocaleShow", func(t *testing.T) {
+		localeShowOpts := phrase.LocaleShowOpts{}
+		resp, httpRes, err := apiClient.LocalesApi.LocaleShow(context.Background(), "project_id_example", "locale_id", &localeShowOpts)
+		requestUrl := httpRes.Request.URL
+		
+
+		require.Nil(t, err)
+		require.NotNil(t, resp)
+		assert.Equal(t, 200, httpRes.StatusCode)
+		assert.Equal(t, "/projects/project_id_example/locales/locale_id", requestUrl.Path)
+		assert.Equal(t, "GET", httpRes.Request.Method)
+	})
+
+	t.Run("Test LocalesApiService LocaleUpdate", func(t *testing.T) {
+		localeUpdateOpts := phrase.LocaleUpdateOpts{}
+		localeUpdateParameters := phrase.LocaleUpdateParameters{}
+		resp, httpRes, err := apiClient.LocalesApi.LocaleUpdate(context.Background(), "project_id_example", "locale_id", localeUpdateParameters, &localeUpdateOpts)
+		requestUrl := httpRes.Request.URL
+		
+
+		require.Nil(t, err)
+		require.NotNil(t, resp)
+		assert.Equal(t, 200, httpRes.StatusCode)
+		assert.Equal(t, "/projects/project_id_example/locales/locale_id", requestUrl.Path)
+		assert.Equal(t, "PATCH", httpRes.Request.Method)
+	})
+
+	t.Run("Test LocalesApiService LocalesList", func(t *testing.T) {
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// Send the mock response
+			response := `[{"foo": "bar"}]`
```

Comment:
can we also return a proper format and actually assert that returned locales are deserialized properly?